### PR TITLE
Fix changing notebook Python intepreter also changing console

### PR DESCRIPTION
**Pull request by seeM**
Addresses #5305. I also added unit tests for the Python runtime session class. They're not necessary for this specific change, but the setup will be useful moving forward.

### CODE REVIEW

File: extensions/positron-python/src/test/mocks/pst/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2023-2024 Posit Software, PBC. All rights reserved.
```
Comment (nstrayer) [inline extensions/positron-python/src/test/mocks/pst/index.ts:2]: nit, does this need to have 2023 in it as well?

File: extensions/positron-python/src/test/vscode-mock.ts [hunk 2]
```diff
@@ -174,3 +176,7 @@ mockedVSCode.LogLevel = vscodeMocks.LogLevel;
 (mockedVSCode as any).ProtocolTypeHierarchyItem = vscodeMocks.vscMockExtHostedTypes.ProtocolTypeHierarchyItem;
 (mockedVSCode as any).CancellationError = vscodeMocks.vscMockExtHostedTypes.CancellationError;
 (mockedVSCode as any).LSPCancellationError = vscodeMocks.vscMockExtHostedTypes.LSPCancellationError;
+
```
Comment (nstrayer) [inline extensions/positron-python/src/test/vscode-mock.ts:179]:
```suggestion
```

**Review by nstrayer — DISMISSED** [review]
Tested locally and works as advertised. I couldn't actually figure out how to run the unit test.

**Comment by nstrayer** [comment]
Oh I think we need to add 
```
        // ...
	{
		label: 'positron-python',
		workspaceFolder: path.join(os.tmpdir(), `positron-python-${Math.floor(Math.random() * 100000)}`),
		mocha: { timeout: 60_000 }
	}
	// ...
```
to `.vscode-test.js` to run the tests.

File: extensions/positron-python/src/test/mocks/pst/index.ts (hunk 1 above)
Comment (seeM) [inline extensions/positron-python/src/test/mocks/pst/index.ts:2]: Nope, bad copy paste.
```suggestion
 *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
```

**Comment by seeM** [comment]
Ah, the Python extension's tests are a bit complex and run differently to the others. You can run the unit tests with `yarn test:unittests` from the extension folder.

**Pull request merged**